lab_3/main.py

```diff
@@ -59,42 +108,79 @@ def encode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
     :param corpus: a tuple of sentences
     :return: a tuple of the encoded sentences
     """
-    pass
+    if not isinstance (corpus, tuple) or type(storage) in [int, str, list, tuple, dict, float, bool] or \
```

Comment:
`type(storage) in [int, str, list, tuple, dict, float, bool]` is over complicated.
You should consider the *appropriate* types, not the inappropriate.